src/easylink/pipeline_graph.py

```diff
@@ -0,0 +1,72 @@
+import itertools
+from pathlib import Path
+from typing import Any, List, Tuple
+
+import networkx as nx
+from networkx import MultiDiGraph
+
+from easylink.configuration import Config
+from easylink.implementation import Implementation
+
+
+class PipelineGraph(MultiDiGraph):
+    def __init__(self, config):
+        super().__init__()
+        self._create_graph(config)
+
+    def _create_graph(self, config: Config) -> None:
+        self.config = config
+        self.add_node("input_data")
+        prev_nodes = None
+
+        for step in config.schema.steps:
+            step_graph = step.get_subgraph(config)
+            step_source_nodes = [node for node, deg in step_graph.in_degree() if deg == 0]
```

Comment:
Maybe add comments throughout here to guide readers who aren't familiar with MultiDiGraph.